scipy/io/wavfile.py

```diff
@@ -243,7 +244,11 @@ def read(filename, mmap=False):
             chunk_id = fid.read(4)
 
             if not chunk_id:
-                raise ValueError("Unexpected end of file.")
+                if data_chunk_received:
+                    # End of file but data successfully read
+                    break
```

Comment:
Instead of just `break`ing here, it might be nice to emit a warning (with appropriate `stacklevel`) saying that only x out of y frames expected from the header could be read

Reply:
I've included a warning following an example from elsewhere in the file, although I haven't set the `stacklevel` and I'm not convinced my wording of the warning is appropriate.

Reply:
The wording looks fine. Probably `stacklevel=2` would be the appropriate level, since this is emitted directly from the `read` function.